src/main/kotlin/kotliquery/package.kt

```diff
@@ -26,6 +26,18 @@ fun sessionOf(dataSource: DataSource, returnGeneratedKey: Boolean = false): Sess
     return Session(Connection(dataSource.connection), returnGeneratedKey)
 }
 
+fun <T> sessionOf(dataSource: DataSource, returnGeneratedKey: Boolean = false, operation: (Session) -> T): T {
```

Comment:
Using the same name `sessionOf` here is very confusing. Is there a better name for this? I think `withSession` may be okay but I'm not so confident.

Reply:
How, about changing it to extension functions:
```kotlin
fun <T> DataSource.session(...
```
```kotlin
fun <T> DataSource.transaction(...
```

Usage would be now:
```kotlin
val db: DataSource
db.session {

}

db.transaction {

}
```
I agree that reusing the name `sessionOf` might be confusing. Furthermore, when I used it in my project I noticed that IDE has slight problems with finding proper imports.

Reply:
How about having `use` method for `AutoCloseable` as with `kotlin.io.use`? 
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html

Or, I haven't checked in detail yet but if implementing both `Closeable` and `AutoCloseable`, we can just use `kotlin.io.use`.

Reply:
That would also be awesome too - maybe even better. Though, I would still like to see a shorthand function like the `transactionOf` above.